src/error.rs

```diff
@@ -188,14 +188,44 @@ where
 {
     fn from(err: E) -> Self {
         Self {
-            kind: Arc::new(err.into()),
+            kind: err.into(),
             labels: Vec::new(),
         }
     }
 }
 
+impl From<bson::de::Error> for ErrorKind {
+    fn from(err: bson::de::Error) -> Self {
+        Self::BsonDecode(Arc::new(err))
+    }
+}
+
+impl From<bson::ser::Error> for ErrorKind {
+    fn from(err: bson::ser::Error) -> Self {
+        Self::BsonEncode(Arc::new(err))
+    }
+}
+
+impl From<std::io::Error> for ErrorKind {
+    fn from(err: std::io::Error) -> Self {
+        Self::Io(Arc::new(err))
+    }
+}
+
+impl From<std::io::ErrorKind> for ErrorKind {
+    fn from(err: std::io::ErrorKind) -> Self {
+        Self::Io(Arc::new(err.into()))
+    }
+}
+
+impl From<tokio::time::error::Elapsed> for ErrorKind {
```

Comment:
I think this error and the async-std equivalent we can just drop, since we convert them both to `std::io::Error`s anyways. Alternatively, we could just implement a custom clone for this (possibly via `Derivative` or something) since [`tokio::time::error::Elapsed` is a unit struct](https://docs.rs/tokio/0.3.2/src/tokio/time/error.rs.html#39).

Reply:
Sounds good; I've updated it to remove those variants